lib/collection/src/shards/shard_holder/mod.rs

```diff
@@ -63,6 +63,14 @@ pub struct ShardHolder {
 pub type LockedShardHolder = RwLock<ShardHolder>;
 
 impl ShardHolder {
+    pub async fn trigger_optimizers(&self) -> CollectionResult<()> {
+        for shard in self.shards.values() {
+            shard.trigger_optimizers().await?;
```

Comment:
I'm not sure if we really need to call this for each shard to be honest.

Reply:
Yes.

I mistakenly described the problem at collection level. But you're right. Each shard has their own optimizers so we need to trigger all of them.